pkg/services/dashboards/service/integration_test/dashboard_k8s_test.go

```diff
@@ -0,0 +1,1680 @@
+			editorUserID := ctx.EditorUser.Identity.GetUID()
+			setResourceUserPermission(t, ctx, ctx.AdminUser, "folders", folderUID, editorUserID, dashboardaccess.PERMISSION_VIEW)
+
+			// Test that editor can view dashboards in the folder (should succeed)
+			t.Run("Editor can view dashboards in restricted folder", func(t *testing.T) {
+				// Create a dashboard in the folder with admin
+				dash, err := createDashboard(t, adminClient, "Admin Dashboard in Restricted Folder", &folderUID, nil)
+				require.NoError(t, err)
+				require.NotNil(t, dash)
+
+				// Editor should be able to view the dashboard (view permission)
+				viewedDash, err := editorClient.Resource.Get(context.Background(), dash.GetName(), v1.GetOptions{})
+				require.NoError(t, err)
+				require.NotNil(t, viewedDash)
+
+				// Clean up
+				err = adminClient.Resource.Delete(context.Background(), dash.GetName(), v1.DeleteOptions{})
+				require.NoError(t, err)
+			})
+
+			// Test that editor cannot create a dashboard in the folder (should fail)
+			t.Run("Editor cannot create dashboard in restricted folder", func(t *testing.T) {
+				// Try to create a dashboard in the folder with editor
+				_, err = createDashboard(t, editorClient, "Editor Dashboard in Restricted Folder", &folderUID, nil)
+				require.Error(t, err, "Should not be able to create dashboard with only VIEW permission")
+			})
+
+			// Test that editor cannot update a dashboard in the folder (should fail)
+			t.Run("Editor cannot update dashboard in restricted folder", func(t *testing.T) {
+				// Create a dashboard in the folder with admin
+				dash, err := createDashboard(t, adminClient, "Dashboard to Update in Restricted Folder", &folderUID, nil)
+				require.NoError(t, err)
+				require.NotNil(t, dash)
+
+				// Editor should not be able to update the dashboard (only has view permission)
+				_, err = updateDashboard(t, editorClient, dash, "Updated by Editor in Restricted Folder")
+				require.Error(t, err, "Should not be able to update dashboard with only VIEW permission")
+
+				// Clean up
+				err = adminClient.Resource.Delete(context.Background(), dash.GetName(), v1.DeleteOptions{})
+				require.NoError(t, err)
+			})
+
+			// Now change to EDIT permissions and verify behavior changes
+			t.Run("Change to EDIT permissions", func(t *testing.T) {
+				// Change permissions for the editor to EDIT
+				setResourceUserPermission(t, ctx, ctx.AdminUser, "folders", folderUID, editorUserID, dashboardaccess.PERMISSION_EDIT)
+
+				// Test that editor can now create a dashboard in the folder (should succeed)
+				t.Run("Editor can now create dashboard in folder", func(t *testing.T) {
+					// Create a dashboard in the folder with editor
+					dash, err := createDashboard(t, editorClient, "Editor Dashboard with EDIT Permission", &folderUID, nil)
+					require.NoError(t, err)
+					require.NotNil(t, dash)
+
+					// Verify it was created in the correct folder
+					spec := dash.Object["spec"].(map[string]interface{})
```

Comment:
The folder UID should also be set as an annotation on the object, so may be good to check that instead (or both?)

Reply:
Done